src/Actions/Product/CreateProduct.php

```diff
@@ -54,6 +56,13 @@ public function performAction(): Product
             $product->prices()->createMany($prices);
         }
 
+        if ($productCrossSellings) {
+            foreach ($productCrossSellings as $productCrossSelling) {
+                $productCrossSelling['product_id'] = $product->id;
+                CreateProductCrossSelling::make($productCrossSelling)->checkPermission()->validate()->execute()->id;
```

Comment:
if you want to check the permission do this beforehand, outside the foreach loop. In this case for each productCrossSelling the same permission is checked although its extremly unlikely that the users permissions do change while looping. And even if they change, this can cause an unexpected behaviour for the user.